src/couch_mrview/src/couch_mrview_debug.erl

```diff
@@ -48,3 +78,36 @@ view_signature(DbName, DDocName) ->
     {ok, DDoc} = couch_db:open_doc_int(Db, <<"_design/", DDocName/binary>>, []),
     {ok, IdxState} = couch_mrview_util:ddoc_to_mrst(DDocName, DDoc),
     couch_util:to_hex(IdxState#mrst.sig).
+
+view_state(Pid) ->
+    {ok, IdxState} = couch_index:get_state(Pid, 0),
+    case IdxState of
+        #mrst{} ->
+            {ok, Info} = couch_index:get_info(Pid),
+            Sig = IdxState#mrst.sig,
+            DbName = IdxState#mrst.db_name,
+            MrView = lists:keyfind(mrview, 1, IdxState#mrst.views),
+            {pending_updates, PendingUpdates} = lists:keyfind(pending_updates, 1, Info),
+            Report = [
```

Comment:
How about making report a map and making `couch_debug:print_report/1` work with maps or updating `view_report/1` to convert from map to list before calling `print_report`?

Reply:
I prefer having `view_report/1` convert from a map to a list mainly because `print_report` uses `bind_value` which expects a list. Having `view_state/1` return a map is better though.